File: src/main/java/com/cloudogu/gitops/infrastructure/git/providers/scmmanager/Permission.java
package com.cloudogu.gitops.infrastructure.git.providers.scmmanager;

import java.util.ArrayList;
import java.util.List;

public record Permission(String name, Role role, boolean groupPermission, List<String> verbs) {

  public Permission(String name, Role role) {
    this(name, role, false, new ArrayList<>());
  }

  public Permission(String name, Role role, boolean groupPermission) {
    this(name, role, groupPermission, new ArrayList<>());
  }

  public Permission {
    verbs = verbs != null ? List.copyOf(verbs) : List.of();
  }

  public enum Role {
    READ,
    WRITE,
    OWNER
  }
}

File: src/main/kotlin/com/cloudogu/gitops/infrastructure/git/providers/scmmanager/Permission.kt
package com.cloudogu.gitops.infrastructure.git.providers.scmmanager

class Permission @JvmOverloads constructor(
    val name: String,
    val role: Role,
    val groupPermission: Boolean = false,
    val verbs: List<String> = emptyList()
) {
    override fun toString(): String {
        return "Permission{name='$name', role=$role, verbs=$verbs, groupPermission=$groupPermission}"
    }

    enum class Role {
        READ, WRITE, OWNER
    }
}
